### PR TITLE
fix: parse ci failure error

### PR DESCRIPTION
**Pull request by F3n67u**
fix https://github.com/nodejs/node-core-utils/issues/639

### CODE REVIEW

**Review by MoLow — APPROVED** [review]
LGTM,
probably best to add a test here? https://github.com/nodejs/node-core-utils/blob/main/test/unit/ci_failure_parser.test.js

**Comment by F3n67u** [comment]
I will add a test for this fix.

**Comment by F3n67u** [comment]
> LGTM, probably best to add a test here? https://github.com/nodejs/node-core-utils/blob/main/test/unit/ci_failure_parser.test.js

of course. I am working on the test. this is why this pr is a draft.😀

**Review by codebytere — APPROVED** [review]

**Comment by F3n67u** [comment]
I added a test for the fix, please take another look.

File: lib/ci/ci_failure_parser.js [hunk 1]
```diff
@@ -155,7 +155,7 @@ const FAILURE_FILTERS = [{
 }, {
   // TODO: match indentation to avoid skipping context with '...'
   filter(ctx, text) {
-    const pattern = /not ok \d+[\s\S]+? {2}\.\.\.\r?\n/mg;
+    const pattern = /^not ok \d+[\s\S]+? {2}\.\.\.\r?\n/mg;
```
Comment (aduh95) [inline lib/ci/ci_failure_parser.js:158]: Following the same logic
```suggestion
    const pattern = /^not ok \d+[\s\S]+? {2}\.\.\.$/mg;
```

Comment (F3n67u) [reply lib/ci/ci_failure_parser.js]: thanks

Comment (F3n67u) [reply lib/ci/ci_failure_parser.js]: Changing `\r?\n` to `$` will make this regex not match `\r?\n` after `...`, this is why the test fails. May I revert this change?

Comment (F3n67u) [reply lib/ci/ci_failure_parser.js]: Reason to revert:

1. after the change, the `reason` of the parse result has `\n` at the end of each line except the last line `...`. I thinks we would better keep consistent: add `\n` to the last line also.
2. avoid potential breaking change, if anywhere consume the reason property.

**Review by aduh95 — APPROVED** [review]

**Comment by F3n67u** [comment]
Update CI reliability job stop working again, please help to merge this pr, Thank you! https://github.com/nodejs/reliability/runs/7587812485?check_suite_focus=true

**Pull request merged**